index.html

```diff
@@ -86,49 +138,211 @@
     // SVG
 
-    svg = d3.select('#chart')
-      .append('svg');
+    chart.svg = d3.select('#chart')
+      .append('svg')
+      .attr("width", width + margin.right + margin.left)
+      .attr("height", height + margin.top + margin.bottom)
+      //.style("background", "lightgrey")
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
 
     // SCALES
 
+    chart.x = d3.scale.log()
+          .base(1.5)
+          .domain(d3.extent(data, function (d) { return d.democratContribution; }))
+          .range([0, width])
+          .nice();
+
+    chart.y = d3.scale.log()
+          .base(1.5)
+          .domain(d3.extent(data, function (d) { return d.republicanContribution; }))
+          .range([height, 0])
+          .nice();
+
+
     // AXES
+    
+    formatAxis = d3.format(".2s");
 
-    // EVENT HANDLERS
+    var xAxis = d3.svg.axis()
+                  .scale(chart.x)
+                  .tickSize(-height)
+                  .orient("bottom")
+                  .tickFormat(formatAxis);
 
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
-      update();
-    });
+    var yAxis = d3.svg.axis()
+                  .scale(chart.y)
+                  .tickSize(-width)
+                  .orient("left")
+                  .tickFormat(formatAxis);
+
+    chart.svg.append("g")
+       .attr("class", "axis") 
+       .attr("transform", "translate(0," + (height) + ")")
+       .call(xAxis)
+       .append("text")
+       .attr("class", "axis-title")
+       .attr("y", margin.bottom)
+       .attr("x", width/2)
+       .style("text-anchor", "middle")
+       .text("Total Contributions to Democrats");
+
+    chart.svg.append("g")
+       .attr("class", "axis") 
+       //.attr("transform", "translate(0,0)")
+       .call(yAxis)
+       .append("text")
+       .attr("class", "axis-title")
+       .attr("transform", "rotate(-90)")
+       .attr("y", -margin.left)
+       .attr("x", -(height/2))
+       .attr("dy", ".71em")
+       .style("text-anchor", "middle")
+       .text("Total Contributions to Republicans");;
+
+    chart.tooltip = d3.tip()
+                      .attr('class','tooltip')
+                      .html(function (d) {return 'Contributions to Democrats: $' + d3.format(",.2f")(d.democratContribution) })
+
+
+
+       chart.update();
+       chart.tooltip(chart.svg); 
   }
 
-  function update() {
+  Chart.prototype.update = function() {
+
+          var chart = this;
+
     // TRANSFORM
     txData = data.slice();
 
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
+
+    if (options.state !== 'all') {
       txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
-      });
+        return d.state === options.state;
+      }); 
     }
 
     // SCATTERPLOT
+  
+
+    var points = chart.svg.selectAll('.point')
+                    .data(txData);
```

Comment:
As we showed in class, you probably want to add a key function here, to prevent one state's points from transitioning to another's, as this would suggest that they are related in some way.